NGitLab.Mock/Project.cs

```diff
@@ -296,6 +297,57 @@ public MergeRequest CreateNewMergeRequest(User user, string branchName, string t
             return mergeRequest;
         }
 
+        public MergeRequest CreateMergeRequest(User user, string title, string description, string targetBranchName, string sourceBranchName, Project sourceProject = null)
+        {
+            var targetProject = this;
+            if (sourceProject is not null && sourceProject != targetProject && sourceProject.ForkedFrom != targetProject)
+                throw new InvalidOperationException("Cannot create a merge request from a source project different from the target project if the former is not a fork of the latter");
+
+            sourceProject ??= this;
+
+            var sourceBranchCommit = sourceProject.Repository.GetBranchTipCommit(sourceBranchName);
+            if (sourceBranchCommit is null)
+                throw new InvalidOperationException($"Branch '{sourceBranchName}' not found in source project");
+
+            var targetBranchCommit = targetProject.Repository.GetBranchTipCommit(targetBranchName);
+            if (targetBranchCommit is null)
+                throw new InvalidOperationException($"Branch '{targetBranchName}' not found in target project");
+
+            // If source project is not target project, fetch the former's source branch into the latter
+            var consolidatedSourceBranchName = sourceBranchName;
+            if (sourceProject != targetProject)
+            {
+                consolidatedSourceBranchName = targetProject.Repository.FetchBranchFromFork(sourceProject, sourceBranchName);
+            }
+
+            var commonCommit = targetProject.Repository.FindMergeBase(targetBranchCommit, sourceBranchCommit);
+            if (commonCommit is null)
+                throw new InvalidOperationException($"Branch '{sourceBranchName}' does not seem to stem from branch '{targetBranchName}'");
+
+            // Determine if there are conflicts, by performing a rebase on a volatile copy of the source branch
+            var volatileBranchName = "volatile-" + Guid.NewGuid().ToString("N");
+            targetProject.Repository.CreateBranch(volatileBranchName, consolidatedSourceBranchName);
+            var hasConflicts = !targetProject.Repository.Rebase(user, volatileBranchName, targetBranchName);
```

Comment:
I don't think this should be checked once. Indeed, the value is obsolete when the source or target branch is updated.

I think this should be a computed property or a method

Reply:
You're right. I made the code somewhat better, recomputing the MR's internal state whenever necessary.